content/tutorials/aws-persistence.textile

```diff
@@ -0,0 +1,305 @@
+
+To follow this tutorial you will need an AWS account. If you don't have an AWS account you can "create one for free":https://aws.amazon.com/free.
+
+h2(#source-code). Tutorial source code
+
+The solution code for this tutorial is "available on Github":https://github.com/ably/ably-messages-s3-tutorial
+
+This repository contains two files:
+
+* @index.html@ - A web page that enables users to subscribe to a channel and publish messages.
+* @lambda_function.py@ - The AWS Lambda function that persists data to S3.
+
+h2(#set-up-user). Create an AWS user
+
+Once you've logged into AWS you need to set up a user capable of using both Lambda and S3. In this tutorial, you will grant high-level policies but you may want to employ stricter policies in a production deployment. You can read more about "AWS authentication":/general/aws-authentication in the Ably docs.
+
+From the "AWS dashboard":http://console.aws.amazon.com/:
+
+1. In the search bar, type "IAM" and press enter.
+
+2. Click the *IAM* link in the search results.
+
+3. Select *Users* from the left hand menu.
+
+4. Click the *Add users* button.
+
+5. Enter @ably-persistence@ as the user name.
+
+6. Check the *Access-key - Programmatic access* checkbox.
+
+7. Click the *Next: permissions* button at the bottom of the page.
+
+8. On the *Set permissions* page:
+    * Select *Attach existing policies directly*.
+    * Type @AWSLambdaRole@ into the search box.
+    * Check the *AWSLambdaRole* checkbox.
+
+9. Click the *Next: Tags* button at the bottom of the page.
+
+10. On the optional Add tags page, click the *Next: Review* buttomn at the bottom of the page.
+
+11. Click @Create user@.
+
+The page displays your new user with its associated *Access key ID* and *Secret access key*. Copy these keys and paste them somewhere: you will need them later in this tutorial.
+
+h2(#set-up-s3). Configure storage with S3
+
+In this tutorial you will use S3 to store a simple plain text file. S3 is one of many scalable storage solutions in the AWS ecosystem but you can apply the principles shown in this tutorial to persist data with other services.
+
+The first step is to set up an S3 bucket to host the file.
+
+From the "AWS dashboard":http://console.aws.amazon.com/:
+
+1. In the console search bar, type "S3" and press enter.
+
+2. Click the *S3* link in the search results.
+
+3. Click *Create bucket*.
+
+4. Enter @ably-message-data@ as the name of your bucket.
+
+5. Click *Create bucket*.
+
+Create the text file and upload it to your new bucket:
+
+6. In your local environment create an empty @test.txt@ file.
+
+7. Select the @ably-message-data@ bucket in the list of buckets.
+
+8. Click the *Upload* button.
+
+9. Click the *Add files* button.
+
+10. Navigate to and select your @test.txt@ file then click *Open*.
+
+11. Click the *Upload* button at the bottom of the page.
```

Comment:
Especially at the end of long sequences of steps like this, there should be some concluding text to confirm to the developer what should have happened, or to orientate them to where they are in the process.